llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -4100,14 +4100,15 @@ static LoadsState canVectorizeLoads(ArrayRef<Value *> VL, const Value *VL0,
     // increases the cost.
     Loop *L = LI.getLoopFor(cast<LoadInst>(VL0)->getParent());
     bool ProfitableGatherPointers =
-        static_cast<unsigned>(count_if(
-            PointerOps,
-            [L](Value *V) { return L && L->isLoopInvariant(V); })) <= Sz / 2 &&
+        L &&
+        count_if(PointerOps, [L](Value *V) { return L->isLoopInvariant(V); }) <=
+            Sz / 2 &&
         Sz > 2;
```

Comment:
Worth moving the Sz > 2 earlier so we avoid the count_if entirely if unnecessary?

Reply:
Yes, will do